### DISCUSSION TITLE
Redux devtools - Missing store/instance name

### DISCUSSION THREAD

**Discussion by RowinVanAmsterdam** [pmndrs/zustand#2906] [Bug report]
### Bug Description

When passing a name option in the devtools middleware of Zustand, the specified name should appear as the Redux DevTools instance name. However, instead of reflecting the provided name, the instance often displays a numeric identifier (e.g., 1135658883/1) instead. Occasionally, the correct name is displayed, but this behavior is inconsistent and seems like  unreliable.

```
export const useBearStore = create<BearState>()(
    devtools(
        (set) => ({
            bears: 0,
            increase: (by) => set((state) => ({ bears: state.bears + by }))
        }),
        {
            name: 'bear-storage'
        }
    )
);
```

<img width="757" alt="devtools" src="https://github.com/user-attachments/assets/8bb9e447-adec-4163-8499-a316a88b117c" />

Additional info:
    "next": "15.1.0",
    "react": "^19.0.0",
    "react-dom": "^19.0.0",
    "zustand": "^5.0.2"

### Reproduction Link

https://github.com/RowinVanAmsterdam/ts-devtools-error
or
https://stackblitz.com/edit/stackblitz-starters-91joccqf?file=app%2Fpage.tsx

**Comment by dbritto-dev** [comment]
@RowinVanAmsterdam would you mind creating a minimal repro on stackblitz?

**Reply by RowinVanAmsterdam** [reply]
@dbritto-dev Sure! Just updated the post with stackblitz link: https://stackblitz.com/edit/stackblitz-starters-91joccqf?file=app%2Fpage.tsx

I hope it's working, since I do not see the store always in the devtools now with Stackblitz. Again, sometimes it seems to work and other times it doesn't.

**Reply by Mochamad-Rizky** [reply]
> @dbritto-dev Sure! Just updated the post with stackblitz link: https://stackblitz.com/edit/stackblitz-starters-91joccqf?file=app%2Fpage.tsx
> 
> I hope it's working, since I do not see the store always in the devtools now with Stackblitz. Again, sometimes it seems to work and other times it doesn't.

I got the same issue in here, the name of store wasn't showing to the redux devtools

**Reply by mkf62** [reply]
Any update on this? It seems whatever store is initialized on load of the application is the one that doesn't get named properly for me, but as my other stores become active as I update stuff in their state, they show up in the list with the names I have assigned.

**Reply by Dannymx** [reply]
Getting same issue here: when setting name to a store with devtools, I get random numbers in the Redux Extension, and I can't see the other stores in my application either.

**Comment by mustafaabobakr** [comment]
any news here 👀

**Comment by dbritto-dev** [comment]
You need to follow this -> https://github.com/pmndrs/zustand/pull/3333
